mono/mini/aot-compiler.c

```diff
@@ -13471,7 +13482,11 @@ emit_aot_image (MonoAotCompile *acfg)
 			acfg->tmpbasename = g_build_filename (temp_path, "temp", NULL);
 			acfg->tmpfname = g_strdup_printf ("%s.s", acfg->tmpbasename);
 			acfg->llvm_sfile = g_strdup_printf ("%s-llvm.s", acfg->tmpbasename);
-			acfg->llvm_ofile = g_strdup_printf ("%s-llvm." AS_OBJECT_FILE_SUFFIX, acfg->tmpbasename);
+
+			if (acfg->aot_opts.static_link)
+				acfg->llvm_ofile = g_strdup_printf (acfg->aot_opts.llvm_outfile);
```

Comment:
Why is this needed ?

Reply:
If I specify the out file for the LLVM object (--llvm-outfile), it will still be generated in the temp folder using the tempbase name. If I specify the out file for the none LLVM object (--outfile) it will be generated as specified by the argument. For none static build this is not a problem since the shared library will be the outfile (the rest will only be temporary artifacts anyway), but for static build, it should be placed as specified in llvm-outfile.

Reply:
We are using these options in ios in static mode, wondering why that works then.

Reply:
OK, that is strange since it will then end up in a temp folder using the default naming schema. Lets do like this, I can keep that change downstream for the platform that currently needs it and remove it from this PR for now to reduce risk of regression.

Reply:
Removed that change from this PR.